tools/cluster/cluster.go

```diff
@@ -308,6 +324,73 @@ func (clu *Cluster) start(dataPath string) error {
 	return nil
 }
 
+func (clu *Cluster) KillNode(nodeIndex int) error {
+	if nodeIndex >= len(clu.waspCmds) {
+		return fmt.Errorf("[cluster] Wasp node with index %d not found", nodeIndex)
+	}
+
+	process := clu.waspCmds[nodeIndex]
+
+	if process != nil {
+		err := process.Process.Kill()
+
+		if err == nil {
+			clu.waspCmds[nodeIndex] = nil
+		}
+
+		return err
+	}
+
+	return nil
+}
+
+func (clu *Cluster) FreezeNode(nodeIndex int) error {
+	if nodeIndex >= len(clu.waspCmds) {
+		return fmt.Errorf("[cluster] Wasp node with index %d not found, active processes: %v", nodeIndex, len(clu.waspCmds))
+	}
+
+	process := clu.waspCmds[nodeIndex]
+
+	err := process.Process.Signal(syscall.SIGSTOP)
+
+	return err
+}
+
+func (clu *Cluster) UnfreezeNode(nodeIndex int) error {
+	if nodeIndex >= len(clu.waspCmds) {
+		return fmt.Errorf("[cluster] Wasp node with index %d not found", nodeIndex)
+	}
+
+	process := clu.waspCmds[nodeIndex]
+
+	err := process.Process.Signal(syscall.SIGCONT)
+
+	return err
+}
+
+func (clu *Cluster) RestartNode(nodeIndex int) error {
+	if nodeIndex >= len(clu.waspCmds) {
+		return fmt.Errorf("[cluster] Wasp node with index %d not found", nodeIndex)
+	}
+
+	initOk := make(chan bool, 1)
+
+	cmd, err := clu.startServer("wasp", waspNodeDataPath(clu.DataPath, nodeIndex), nodeIndex, initOk)
+	if err != nil {
+		return err
+	}
+
+	select {
+	case <-initOk:
+	case <-time.After(10 * time.Second):
+		return fmt.Errorf("Timeout starting wasp nodes\n") //nolint:revive
```

Comment:
I think `xerrors` is prefered (instead of fmt.Errorf)
```suggestion
		return xerrors.Errorf("Timeout starting wasp nodes\n")
```

Reply:
Everywhere or just in such functions? Then I replace that in the whole cluster class.

Reply:
I think everywhere, but there are still a low of `fmt.Errorf` across the repo. I'm not sure what's the reason for this decision, just that's what we're using 😄 .
cc: @dessaya

Reply:
It's preferred because when the error is printed, using `xerrors` will print the file:line info and all wrapped errors (when using `%w`), so you get something like a stack trace of wrapped errors.